common/persistence/serialization/blob.go

```diff
@@ -214,13 +215,20 @@ func ReplicationVersionsFromBlob(b []byte, proto string) (*persistenceblobs.Repl
 	return result, proto3Decode(b, proto, result)
 }
 
-func ChecksumToBlob(info *persistenceblobs.Checksum) (commonpb.DataBlob, error) {
-	return proto3Encode(info)
+func ChecksumToBlob(checksum *persistenceblobs.Checksum) (commonpb.DataBlob, error) {
+	if checksum == nil {
+		checksum = &persistenceblobs.Checksum{}
+	}
+	return proto3Encode(checksum)
 }
 
 func ChecksumFromBlob(b []byte, proto string) (*persistenceblobs.Checksum, error) {
 	result := &persistenceblobs.Checksum{}
-	return result, proto3Decode(b, proto, result)
+	err := proto3Decode(b, proto, result)
+	if err != nil || result.GetFlavor() == enumsspb.CHECKSUM_FLAVOR_UNSPECIFIED {
```

Comment:
return an err if `result.GetFlavor() == enumsspb.CHECKSUM_FLAVOR_UNSPECIFIED`?

Reply:
This is basically a check for empty `result` struct. Checksum used to be non-pointers and I had to convert them to pointers everywhere because proto structs use pointers. But DB field doesn't allow NULL so I convert nil to `&persistenceblobs.Checksum{}` few lines above and this empty object goes to blob. Here it converts back from empty struct to `nil`. It is not an error case.

Reply:
plz put some comment here explain the behavior, since this is not a common behavior